### ISSUE TITLE
Allow changes in config.yaml to be picked up on the fly

### ISSUE THREAD

**Issue by andrerung** [ericaltendorf/plotman#254]
I'd like to be able to change config.yaml while plotman is executing, is there a way to implement that?

Best regards
André Rung

**Comment by jjhesk** [comment]
its already done that

**Comment by andrerung** [comment]
I've just tried to enable - tmpdir_max_jobs but it doesn't seem to be picked up. I still have 2 jobs running on that tempdir, which is the max configured.
Am I doing anything wrong?

        # Optional: Allows overriding some characteristics of certain tmp        # directories. This contains a map of tmp directory names to        # attributes. If a tmp directory and attribute is not listed here,        # it uses the default attribute setting from the main configuration.        #        # Currently support override parameters:              - tmpdir_max_jobs        tmp_overrides:                # In this example, /mnt/tmp/00 is larger than the other tmp                # dirs and it can hold more plots than the default.                "/mnt/mp600/tempdir_4":                        tmpdir_max_jobs: 4


André Rung 

    On Thursday, May 6, 2021, 09:48:10 PM GMT+1, MDCCLXXVI KPT ***@***.***> wrote:  
 
 


its already done that

**Comment by altendky** [comment]
plotman doesn't live update from the config file.  I personally think that plotman should become more of a service that can be monitored and I don't think that really aligns with live updates from config.  At present, the only thing you lose by closing and rerunning plotman is if you have disabled either plotting or archiving.  It's not obvious to me that live updating is that valuable.  Am I missing something?

**Comment by andrerung** [comment]
Let's see if I got what you said, you mean that If I stop the "plotman plot" that is running and re-launch it, I don't lose anything and plotman will pick up the changes in the config file. Is that right?
André Rung 

    On Thursday, May 6, 2021, 11:15:22 PM GMT+1, Kyle Altendorf ***@***.***> wrote:  
 
 


plotman doesn't live update from the config file. I personally think that plotman should become more of a service that can be monitored and I don't think that really aligns with live updates from config. At present, the only thing you lose by closing and rerunning plotman is if you have disabled either plotting or archiving. It's not obvious to me that live updating is that valuable. Am I missing something?

**Comment by altendky** [comment]
Yes.  I usually use `plotman interactive` at this point, but the same idea applies. 
 The ability to close and restart is very much an explicit feature.

**Comment by VilleSalonen** [comment]
To clarify there is at least some settings which are picked up on the fly. I added a new tmp directory to config and it was picked up immediately as I expected. I was so discouraged by this issue before that I didn't even try it myself until asking around. Is there any documentation which settings are automatically picked up and which are not?

**Comment by altendky** [comment]
plotman doesn't reload on the fly.  Perhaps you were running `plotman plot`, changed the config, then ran `plotman interactive` and interactive was respecting the new settings and launching plots?

**Comment by VilleSalonen** [comment]
I was running `plotman interactive`. I actually thought interactive mode was just a read-only view to actual plotting done by `plotman plot`. Thank you for your answer.

**Comment by altendky** [comment]
Yeah, presently there's overlap.  Many people have been confused by this.  One of the many things on the long list is to make a stronger distinction between the operational side and the monitoring side.
